core/data/src/main/java/com/google/samples/apps/nowinandroid/core/data/repository/OfflineFirstNewsRepository.kt

```diff
@@ -72,15 +74,25 @@ class OfflineFirstNewsRepository @Inject constructor(
             },
             modelDeleter = newsResourceDao::deleteNewsResources,
             modelUpdater = { changedIds ->
+                val userData = niaPreferencesDataSource.userData.first()
+                val hasOnBoarded = userData.shouldHideOnboarding
+                val followedTopicIds = userData.followedTopics
+
                 // TODO: Make this more efficient, there is no need to retrieve populated
                 //  news resources when all that's needed are the ids
-                val existingNewsResourceIds = newsResourceDao.getNewsResources(
-                    useFilterNewsIds = true,
-                    filterNewsIds = changedIds.toSet(),
-                )
-                    .first()
-                    .map { it.entity.id }
-                    .toSet()
+                val existingFollowedChangedNewsResourceIds = when {
```

Comment:
Consider using `if` instead of `when` for a boolean check

Reply:
Since there is an assignment being done, the `when` seemed a little easier to read since the `if` statement had twice the number of braces.